diff --git a/ggml/src/CMakeLists.txt b/ggml/src/CMakeLists.txt
index 84101c3..92a5fe5 100644
--- a/ggml/src/CMakeLists.txt
+++ b/ggml/src/CMakeLists.txt
@@ -202,7 +202,7 @@ if (GGML_BACKEND_DL AND NOT BUILD_SHARED_LIBS)
     message(FATAL_ERROR "GGML_BACKEND_DL requires BUILD_SHARED_LIBS")
 endif()
 
-add_library(ggml-base
+add_library(ggml-base STATIC
             ../include/ggml.h
             ../include/ggml-alloc.h
             ../include/ggml-backend.h
