diff --git a/scripts/get_cmake_vars/CMakeLists.txt b/scripts/get_cmake_vars/CMakeLists.txt
index 7e5ddbda..3d5be8e2 100644
--- a/scripts/get_cmake_vars/CMakeLists.txt
+++ b/scripts/get_cmake_vars/CMakeLists.txt
@@ -12,7 +12,7 @@ endif()
 
 set(VCPKG_LANGUAGES "C;CXX" CACHE STRING "Languages to enables for this project")
 if(VCPKG_ENABLE_OBJC)
-    list(APPEND VCPKG_LANGUAGES "OBJC")
+    list(APPEND VCPKG_LANGUAGES "OBJC" "OBJCXX")
 endif()
 
 project(get_cmake_vars LANGUAGES ${VCPKG_LANGUAGES})
