diff --git a/ports/pango/portfile.cmake b/ports/pango/portfile.cmake
index df2cf032c9..7ca6990b7f 100644
--- a/ports/pango/portfile.cmake
+++ b/ports/pango/portfile.cmake
@@ -19,8 +19,16 @@ else()
     list(APPEND OPTIONS_RELEASE -Dintrospection=disabled)
 endif()
 
+# <TechSmith Customizations>
+set(LANGUAGES C CXX)
+if(VCPKG_TARGET_IS_OSX OR VCPKG_TARGET_IS_IOS)
+    list(APPEND LANGUAGES OBJC OBJCXX)
+endif()
+# </TechSmith Customizations>
+
 vcpkg_configure_meson(
     SOURCE_PATH "${SOURCE_PATH}"
+    LANGUAGES ${LANGUAGES}
     OPTIONS
         -Ddocumentation=false
         -Dman-pages=false
