diff --git a/ConfigureChecks.cmake b/ConfigureChecks.cmake
index 73a1a4c2..537c68e1 100644
--- a/ConfigureChecks.cmake
+++ b/ConfigureChecks.cmake
@@ -35,7 +35,7 @@ CONFIGURE_FILE(config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config.h)
 
 ADD_DEFINITIONS(-DHAVE_CONFIG_H)
 
-INCLUDE(FindPythonInterp)
+INCLUDE(FindPython3)
 IF (NOT PYTHON_EXECUTABLE)
     MESSAGE(FATAL_ERROR "Could not find Python executable")
 ENDIF (NOT PYTHON_EXECUTABLE)
